sdk/avalon_sdk/ethereum/ethereum_worker_registry.py

```diff
@@ -0,0 +1,276 @@
+        """
+        if config["ethereum"]["worker_registry_contract_file"] is None:
+            logging.error("Missing worker registry contract file path!!")
+            return False
+        if config["ethereum"]["worker_registry_contract_address"] is None:
+            logging.error("Missing worker registry contract address!!")
+            return False
+        return True
+
+    def __initialize(self, config):
+        """
+        Initialize the parameters from config to instance variables.
+        """
+        self.__eth_client = EthereumWrapper(config)
+        tcf_home = environ.get("TCF_HOME", "../../../")
+        contract_file_name = tcf_home + "/" + \
+            config["ethereum"]["worker_registry_contract_file"]
+        contract_address = \
+            config["ethereum"]["worker_registry_contract_address"]
+        self.__contract_instance = self.__eth_client.get_contract_instance(
+            contract_file_name, contract_address
+        )
+
+    def worker_register(
+            self, worker_id, worker_type, org_id, application_ids, details):
+        """
+        Register new worker with details of worker
+        """
+        if (self.__contract_instance is not None):
+            if not is_valid_hex_str(
+                    binascii.hexlify(worker_id).decode("utf8")):
+                logging.info("Invalid worker id {}".format(worker_id))
+                return construct_json_resp_message(
+                    "failed", "Invalid worker id {}".format(worker_id))
+            if not isinstance(worker_type, WorkerType):
+                logging.info("Invalid workerType {}".format(worker_type))
+                return construct_json_resp_message(
+                    "failed", "Invalid workerType {}".format(worker_type))
+            if not is_valid_hex_str(binascii.hexlify(org_id).decode("utf8")):
+                logging.info("Invalid organization id {}".format(org_id))
+                return construct_json_resp_message(
+                    "failed", "Invalid organization id {}".format(org_id))
+            for aid in application_ids:
+                if not is_valid_hex_str(binascii.hexlify(aid).decode("utf8")):
+                    logging.info("Invalid application id {}".format(aid))
+                    return construct_json_resp_message(
+                        "failed", "Invalid application id {}".format(aid))
+            if details is not None:
+                is_valid = validate_details(details)
+                if is_valid is not None:
+                    return construct_json_resp_message("failed", is_valid)
+
+            txn_hash = self.__contract_instance.functions.workerRegister(
+                worker_id, worker_type.value, org_id, application_ids,
+                details).buildTransaction(
+                    self.__eth_client.get_transaction_params()
+                )
+            tx = self.__eth_client.execute_transaction(txn_hash)
```

Comment:
execute_transaction() function returns json status response. So better to name variable "tx" something like  "tx_status" to make the variable name look more meaningful

Reply:
Done